pkg/subctl/cmd/diagnose/globalnet.go

```diff
@@ -0,0 +1,162 @@
+	status := cli.NewStatus()
+
+	if cluster.Submariner == nil {
+		status.Start(cmd.SubmMissingMessage)
+		status.EndWith(cli.Warning)
+
+		return true
+	}
+
+	status.Start("Checking Globalnet configuration")
+
+	if cluster.Submariner.Spec.GlobalCIDR != "" {
+		checkClusterGlobalEgressIps(cluster, status)
+		checkGlobalEgressIps(cluster, status)
+		checkGlobalIngressIps(cluster, status)
+
+		if status.HasFailureMessages() {
+			status.EndWith(cli.Failure)
+			return false
+		}
+
+		status.EndWithSuccess("Globalnet is enabled and properly configured")
+	} else {
+		status.EndWithSuccess("Globalnet is disabled")
+	}
+
+	return true
+}
+
+func checkClusterGlobalEgressIps(cluster *cmd.Cluster, status *cli.Status) {
+	clusterGlobalEgress, err := cluster.SubmClient.SubmarinerV1().ClusterGlobalEgressIPs(
+		corev1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		status.QueueFailureMessage(fmt.Sprintf("Error obtaining the clusterGlobalEgressIPs cr: %v", err))
+		return
+	}
+
+	if len(clusterGlobalEgress.Items) != 1 {
+		status.QueueFailureMessage(fmt.Sprintf("Number (%d) of clusterGlobalEgress resources != 1", len(clusterGlobalEgress.Items)))
+		return
+	}
+
+	if clusterGlobalEgress.Items[0].Name != constants.ClusterGlobalEgressIPName {
+		status.QueueFailureMessage(fmt.Sprintf("clusterGlobalEgress name (%s) not equals to default name", clusterGlobalEgress.Items[0].Name))
+		return
+	}
+}
+
+func checkGlobalEgressIps(cluster *cmd.Cluster, status *cli.Status) {
+	globalEgressIps, err := cluster.SubmClient.SubmarinerV1().GlobalEgressIPs(
+		corev1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		status.QueueFailureMessage(fmt.Sprintf("Error obtaining GlobalEgressIPs resources: %v", err))
+		return
+	}
+
+	for i := range globalEgressIps.Items {
+		if *globalEgressIps.Items[i].Spec.NumberOfIPs != len(globalEgressIps.Items[i].Status.AllocatedIPs) {
+			status.QueueFailureMessage(fmt.Sprintf("Error GlobalEgressIPs(%s) NumberOfIPs != AllocatedIPs", globalEgressIps.Items[i].Name))
+			return
+		}
+	}
+}
+
+func checkGlobalIngressIps(cluster *cmd.Cluster, status *cli.Status) {
+	mcsClient, err := mcsClientset.NewForConfig(cluster.Config)
+	if err != nil {
+		status.QueueFailureMessage(fmt.Sprintf("Error obtaining mcs client: %v", err))
+		return
+	}
+
+	serviceExports, err := mcsClient.MulticlusterV1alpha1().ServiceExports(corev1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		status.QueueFailureMessage(fmt.Sprintf("Error obtaining ServiceExport resources: %v", err))
+		return
+	}
+
+	for i := range serviceExports.Items {
+		ns := serviceExports.Items[i].GetNamespace()
+		name := serviceExports.Items[i].GetName()
+
+		svc, err := cluster.KubeClient.CoreV1().Services(ns).Get(context.TODO(), name, metav1.GetOptions{})
+		if err != nil {
+			status.QueueFailureMessage(fmt.Sprintf("Error matching Service resource for exported service(%s,%s).%v", ns, name, err))
+			return
+		}
```

Comment:
The `ServiceExport` can exist without the `Service` so emit a warning.

**Note**: `import apierrors "k8s.io/apimachinery/pkg/api/errors"`

```suggestion
		if apierrors.IsNotFound(err)  {
		        status.Warning("No matching Service resource found for exported service \"%s/%s\"", ns, name)
			continue
		}
		
		if err != nil {
		    status.Failure("Error retrieving Service: %v", err)
		    return false
		}
```